iceoryx_hoofs/test/moduletests/test_posix_shared_memory_object.cpp

```diff
@@ -82,6 +82,21 @@ TEST_F(SharedMemoryObject_Test, CTorOpenNonExistingSharedMemoryObject)
     EXPECT_THAT(sut.has_error(), Eq(true));
 }
 
+TEST_F(SharedMemoryObject_Test, CTorCreateReadOnlySharedMemoryObject)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "80684160-b243-4ca1-b285-118d2ef36108");
+    auto sut = iox::posix::SharedMemoryObjectBuilder()
+                   .name("readOnlyShmMem")
+                   .memorySizeInBytes(100)
+                   .accessMode(iox::posix::AccessMode::READ_ONLY)
+                   .openMode(iox::posix::OpenMode::PURGE_AND_CREATE)
+                   .create();
+
+    // TODO(tobias.stark): Check that the right error is returned, not just any error.
```

Comment:
I think when the shared memory is created read only the `ftruncate` fails which resizes the shared memory to 100 bytes. Therefore it should be `SharedMemoryError::INSUFFICIENT_PERMISSIONS`.

What is the output of the test when you run the code? Since the underlying posix call abstraction handles all the system calls you should see an error output from the line where it fails and the errno code.

Reply:
I introduced a new error code and moved this test to the `test_posix_shared_memory.cpp` test, where the error-code from `SharedMemory` is available.